apps/balancer-tools/src/app/apr/(utils)/tokenApr.ts

```diff
@@ -285,3 +321,159 @@ const getImplementationAddress = async (
     return undefined;
   }
 };
+
+async function calculateTokenApr({
+  tokenYield,
+  poolId,
+  chain,
+  tokenAddress,
+  tokenWeight,
+  timeStart,
+  timeEnd,
+}: {
+  tokenYield: number;
+  poolId: string;
+  chain: string;
+  tokenAddress: Address;
+  tokenWeight: number;
+  timeStart: number;
+  timeEnd: number;
+}) {
+  const rangeData = await poolsWithCache.gql(chain).poolSnapshotInRange({
+    poolId,
+    timestamp: generateDateRange(timeStart, timeEnd),
+  });
+
+  let pool: Pool;
+
+  if (rangeData.poolSnapshots.length > 0) {
+    pool = rangeData.poolSnapshots[0].pool;
+  } else {
+    const currentPoolData = await pools.gql(chain).Pool({ poolId });
+    if (!currentPoolData || !currentPoolData.pool) {
+      throw new Error("Pool data is not found or is invalid.");
+    }
+    pool = currentPoolData.pool;
+  }
+
+  if (!pool || !pool.tokens) {
+    return 0;
+  }
+
+  let apr;
+
+  if (
+    pool.poolType === PoolTypeEnum.PHANTOM_STABLE ||
+    (pool.poolType === PoolTypeEnum.WEIGHTED &&
+      (pool.poolTypeVersion ?? 0) >= 2)
+  ) {
+    const isExemptFromYieldProtocolFee = pool?.tokens.find(
+      ({ address }) => address === tokenAddress,
```

Comment:
~this assumes only one of the tokens would be exempt from YieldProtocolFee, but there could be multiple for the same pool, no? If not, I think there should be a comment here explaining why~

nvm, read this quickly and saw we're not finding by `isExemptFromYieldProtocolFee` but just getting the token address that's been passed to the function